Api/src/controllers/userController.js

```diff
@@ -0,0 +1,75 @@
+import {
+  v4 as uuidv4
+} from 'uuid';
+
+let users = [{
+    id: uuidv4(),
+    firstName: "Joseph",
+    lastName: "Ikape",
+    age: 30,
+  },
+  {
+    id: uuidv4(),
+    firstName: "Wesley",
+    lastName: "Philip",
+    age: 25,
+  },
+  {
+    id: uuidv4(),
+    firstName: "Christian",
+    lastName: "Bale",
+    age: 46,
+  }
+];
+
+export const getUsers = (req, res, next) => {
+  res.send(users);
+  next();
+};
+
+export const createUser = (req, res, next) => {
+  const user = {
+    id: uuidv4(),
+    firstName: req.body.firstName,
+    lastName: req.body.lastName,
+    age: req.body.age,
+  }
+
+  users.push(user);
+
+  res.send(`User ${user.firstName} added to the database.`);
+  next();
+};
+
+export const getUser = (req, res) => {
+  const userId = req.params.id;
+  const user = users.find(user => user.id === parseInt(userId));
+  if (!userId) return res.statusCode(404).send("No user with this id");
+
+  res.send(user);
+}
+
+export const updateUser = (req, res) => {
+  const userId = users.find((user) => user.id === req.params.id);
+
+  if (!userId) return res.statusCode(400).send("User does not exist");
+
+  userId.firstName = req.body.firstName;
```

Comment:
What if `req.body.firstName` is empty - i.e. was not requested to be updated by user??

You should only update when there is a change:

 ```suggestion
  userId.firstName = req.body.firstName || userId.firstName;
```